drivers/arm/analog.h

```diff
@@ -17,41 +17,27 @@
 #pragma once
 
 #include "quantum.h"
-#include "ch.h"
-#include <hal.h>
 
-#if !defined(STM32F0XX) && !defined(STM32F3XX)
-#    error "Only STM23F0 and STM32F3 devices have ADC support in QMK at this time."
-#endif
-
-#if !HAL_USE_ADC
-#    error "You need to set HAL_USE_ADC to TRUE in your halconf.h to use the ADC."
-#endif
-
-#if !STM32_ADC_USE_ADC1 && !STM32_ADC_USE_ADC2 && !STM32_ADC_USE_ADC3 && !STM32_ADC_USE_ADC4
-#    error "You need to set one of the 'STM32_ADC_USE_ADCx' settings to TRUE in your mcuconf.h to use the ADC."
-#endif
-
-#if STM32_ADC_DUAL_MODE
-#    error "STM32 ADC Dual Mode is not supported at this time."
-#endif
-
-#if STM32_ADCV3_OVERSAMPLING
-#    error "STM32 ADCV3 Oversampling is not supported at this time."
+#ifdef __cplusplus
+extern "C" {
 #endif
 
 typedef struct {
-    pin_t   pin;
+    uint16_t input;
     uint8_t adc;
-} pin_and_adc;
-#define PIN_AND_ADC(p, a) \
-    (pin_and_adc) { p, a }
+} adc_mux;
+#define TO_MUX(i, a) \
+    (adc_mux) { i, a }
 
 // analogReference has been left un-defined for ARM devices.
 // void analogReference(uint8_t mode);
 
```

Comment:
```suggestion
```
Let's just note in the docs that this function is AVR only.

Reply:
At the moment the docs have the functions broken out into ARM/AVR so this should be covered.